core/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -153,6 +153,9 @@ private fun <T : Any> Class<T>.createEnumSerializer(): KSerializer<T> {
 }
 
 private fun <T : Any> Class<T>.findObjectSerializer(): KSerializer<T>? {
+    // Special case to avoid IllegalAccessException on Java11+ (#2449)
+    // There are no serializable objects in the stdlib anyway.
+    if (this.canonicalName.let { it.startsWith("java.") || it.startsWith("kotlin.") }) return null
     // Check it is an object without using kotlin-reflect
     val field =
         declaredFields.singleOrNull { it.name == "INSTANCE" && it.type == this && Modifier.isStatic(it.modifiers) }
```

Comment:
INSTANCE could also not be public, and therefore the type would not be an object (small chance, but add a check for it being public).
```suggestion
    val field = declaredFields.singleOrNull {
        it.name == "INSTANCE" &&
            it.type == this &&
            Modifier.isStatic(it.modifiers) &&
            Modifier.isPublic(it.modifiers)
    } ?: return null
```

Reply:
Can you provide an example? IIRC Kotlin doesn't produce INSTANCE outside of Kotlin objects and it's hard to produce such code by yourself

Reply:
Basically if someone were to have a Java based singleton such as:
```java
class BreakThings {
    private static BreakThings INSTANCE = BreakThings();

    public static BreakThings getInstance() { return INSTANCE; }
}

Reply:
I think it's ok to fail or have unspecified behavior when asked for serializer for some java class